### PR TITLE
add point tool to GraphTool

### PR DESCRIPTION
**Pull request by Alex-Jordan**
This pairs with https://github.com/openwebwork/pg/pull/610. The intent is to have a point-plotting tool. My most immediate use case is basic algebra with students learning the cartesian coordinate system.

These edits are incomplete. I'm opening this PR to make it easier for @drgrice1  to give me guidance.

My sense is I have the right changes to parserGraphTool.pl (over in `pg`). Mainly the tikz addition. And here in `webwork2`, I made a new point tool icon image, and put that in graphtool.css.

So that leaves graphtool.js. For testing, I'm using the un-minified graphtool.js. (I will need recommendations for minfying later.) My sense is that I added the right things to make a Point graph object, but I'm not feeling secure about that. I copied the Line object and just made what seem to me like the right edits.

That leaves making the PointTool object. For this I copied the FillTool (since unlike the other tools it is a one-click affair). I realize the fill icon section will need to be taken out, but it seems like I've hit a place where I have more fundamental issues to iron out first. 

At this point I get javascript errors if I try to use my new PointTool. The error (pasted below) indicates it is trying to stringify the point but failing.

```
Uncaught TypeError: obj.stringify is not a function
    value https://webwork.pcc.edu/webwork2_files/js/apps/GraphTool/graphtool.js:169
    updateText https://webwork.pcc.edu/webwork2_files/js/apps/GraphTool/graphtool.js:167
    activate https://webwork.pcc.edu/webwork2_files/js/apps/GraphTool/graphtool.js:900
```

(Also I know that the PointTool will not be there by default in the end, but I have it that way for now in development.)

### CODE REVIEW

**Review by drgrice1 — CHANGES_REQUESTED** [review]
I have commented in the code.

I also will submit a pull request to this one with some changes that will make the point tool work.

File: htdocs/js/apps/GraphTool/graphtool.js [hunk 1]
```diff
@@ -315,6 +316,38 @@ function graphTool(containerId, options) {
...
+	Point.prototype.stringify = function() {
+		return [
+			Point.strId,
+			"(" + gt.snapRound(this.definingPts.point.X(), gt.snapSizeX) + "," +
+			gt.snapRound(this.definingPts.point.Y(), gt.snapSizeY) + ")"
+		].join(",");
+	};
+	Point.prototype.fillCmp = function(point) {
+		return 1;
+	};
```
Comment (drgrice1) [inline htdocs/js/apps/GraphTool/graphtool.js:339]: Note that I removed the fillCmp override in my pull request.  There is no need to override the parent class method with a function that does the same thing.

**Comment by drgrice1** [comment]
I added another change that I think fixes those issues.

**Comment by drgrice1** [comment]
That is done by the graph took JavaScript.  We can fix that. I will look at it tomorrow.

**Comment by Alex-Jordan** [comment]
Following up on this. Based on:

https://github.com/openwebwork/pg/blob/37637e2f880bccbbb8781b5d669421849ac81267/lib/Value/AnswerChecker.pm#L1427

and the surrounding code, yes, a list_checker is looking for a score that is the maximum between the size of the correct list and the student's list.

**Comment by drgrice1** [comment]
I just submitted two paired pull requests to your branches for this pull request and for openwebwork/pg#610.  Both versions remove the PointTool from the GraphTool.pl macro and javascript directly, and instead move it into a separate macro that adds to the parserGraphTool.pl macro.  The difference between the two versions is that version 1 loads the javascript from the separate javscript file htdocs/js/apps/GraphTool/pointtool.js, and the other injects the javascript inline from the macro.

The point of this is that I was thinking it would be neater to separate this from the main tools.  Although I am thinking of moving all of the tools into separate js files loaded in this way that the version 1 pull requests do.

Of the two versions I certainly prefer version 1.

This is a suggestion and I would like to know what you think.

There is an example in the added macro file that is a full pg example.

**Comment by drgrice1** [comment]
I just submitted two paired pull requests to your branches for this pull request and for openwebwork/pg#610.  Both versions remove the PointTool from the GraphTool.pl macro and javascript directly, and instead move it into a separate macro that adds to the parserGraphTool.pl macro.  The difference between the two versions is that version 1 loads the javascript from the separate javscript file htdocs/js/apps/GraphTool/pointtool.js, and the other injects the javascript inline from the macro.

The point of this is that I was thinking it would be neater to separate this from the main tools.  Although I am thinking of moving all of the tools into separate js files loaded in this way that the version 1 pull requests do.

Of the two versions I certainly prefer version 1.

This is a suggestion and I would like to know what you think.

There is an example in the added macro file that is a full pg example.

**Comment by drgrice1** [comment]
I just realized my comment was never submitted.  Good think github saves them even if you don't hit the "Comment" button.

**Comment by drgrice1** [comment]
@Alex-Jordan: I hope you don't mind, but I fixed the merge conflicts created by the merge of #1532.  I am working on updating my pull requests to your branch.

**Comment by Alex-Jordan** [comment]
No problem. I hope *you* don't mind my tardiness in tying this all up. I'm into week 2 of a heavy term. I'm feeling like I might have some freedom next week.

**Comment by drgrice1** [comment]
No problem.  I know you are busy with classes right now.  I have the updated pg pull request for you in a branch in my fork. I can help you copy that over to your fork if you need it.

**Comment by drgrice1** [comment]
This pull request is broken again with the merge of #1536, and this time it can not be repaired here.  On the other hand, thanks to that pull request, this can be done as a single pull request to openwebwork/pg.  I have added the javascript, css, and point tool svg to the pointTool branch in my fork.  So whenever you have time, you just need to copy that branch and put it in as a pull request.  I will also update my suggested changes to a separate macro.

**Comment by drgrice1** [comment]
@Alex-Jordan:  I am going to close this pull request.  It will no longer be needed.  With the changes to the webwork/pg structure this can now all be done with a single pull request on the pg side.  I have a branch on my fork that has this all with the git history (at least from the pg side), is cleanly rebased onto the current development branch, and merges without git complaining about the prior revert.  Probably the easiest way to get that branch to you, is for you to add my fork as a remote to your repository, clone the branch, and then push it to your fork.

**Pull request closed without merging**